test/kubernetes_test.go

```diff
@@ -479,8 +840,63 @@ func TestKubernetesIntegrationFallthrough(t *testing.T) {
 	drop 0
     }
 `
-	cases := append(dnsTestCases, dnsTestCasesFallthrough...)
-	doIntegrationTests(t, corefile, cases)
+	doIntegrationTests(t, corefile, dnsTestCasesFallthrough)
+}
+
+func getDynamicEndpoints(Qname, namespace, name string) []dns.RR {
+	rr := []dns.RR{}
+
+	result := strings.Split(getCmdOutput(name, namespace), " ")
+	len_result := len(result)
+
+	if len_result > 1 {
+		for i := 0; i < len_result; i++ {
+			rr = append(rr, test.A(Qname+"    303    IN      A   "+result[i]))
+		}
+	} else {
+		rr = append(rr, test.A(Qname+"    303    IN      A   "+result[0]))
+	}
+
+	return rr
+}
+
+func getSRVrecords(Qname, Qtype, namespace, name string) []dns.RR {
+	rr := []dns.RR{}
+
+	result := strings.Split(getCmdOutput(name, namespace), " ")
+	len_result := len(result)
+
+	if len_result > 1 {
+		for i := 0; i < len_result; i++ {
+			ip := strings.Replace(result[i], ".", "-", -1)
+			if Qtype == "TypeSRV" {
+				rr = append(rr, test.SRV(Qname+"   303    IN    SRV 0 33 1234  "+ip+"."+namespace+"."+name+".svc.cluster.local."))
+			} else {
+				rr = append(rr, test.A(ip+"."+namespace+"."+name+".svc.cluster.local.	0	IN	A	"+result[i]))
+			}
+		}
+	} else {
+		ip := strings.Replace(result[0], ".", "-", -1)
+		rr = append(rr, test.SRV(Qname+"   303    IN    SRV 0 33 1234  "+ip+"."+namespace+"."+name+".svc.cluster.local."))
+	}
+	return rr
+}
+
+func getCmdOutput(name, namespace string) string {
+	var (
+		cmdOut []byte
+		err    error
+	)
+	kctl := os.Getenv("KUBECTL")
+	if kctl == "" {
+		kctl = "kubectl"
+	}
+	cmdArgs := kctl + " -n " + name + " get endpoints " + namespace + " -o jsonpath={.subsets[*].addresses[*].ip}"
+	if cmdOut, err = exec.Command("sh", "-c", cmdArgs).Output(); err != nil {
+		println(os.Stderr, "There was an error running kubectl command: ", err.Error())
+		os.Exit(1)
+	}
+	return string(cmdOut)
```

Comment:
The `bytes` pkg has almost all operations the string package has, there is no need to convert this into a `string`.